oauthenticator/oauth2.py

```diff
@@ -456,6 +456,24 @@ def _logout_redirect_url_default(self):
         """,
     )
 
+    required_scopes = List(
```

Comment:
I think the terminology is that scopes are what you request, and claims is what you get - so this should be required_claims then I think.

A google provided this:

> Scopes are only present in OAuth Access Tokens. The ID Token never contains scopes but instead contains the resolved claims that belong to the issued scopes.

> Scopes are groups of claims.

So if you request the scope access_levels you can get claims gpu_access=true and tpu_access=false sent back.

Reply:
Hmmm is it:
- requested scopes,
- issues scopes (groups of claims),
- and claims being key=values?

Then, should this be required_issued_scopes, or required_claims? If its required claims, should it allow to specify the claim key and/or claim key/values?

Reply:
Ah nevermind, this is not ambiguous, its required (issued) scopes, and oauth2 provides the specific scope key in response - nice!

Reply:
Thanks for digging through this :) I always forget the terminology if I'm away from oauth2 for like, 3 weeks.